sr/comp/mixtape/mixtape.py

```diff
@@ -23,12 +24,33 @@ def __init__(
         playlist: Any,
         audio_controller: AudioController,
         magicq_controller: Optional[MagicqController],
+        obs_studio_controller: Optional[OBSStudioController],
     ) -> None:
         self.root = root
         self.playlist = playlist
         self.audio_controller = audio_controller
         self.exclusivity_groups: Dict[object, subprocess.Popen[bytes]] = {}
         self.magicq_controller = magicq_controller
+        self.obs_studio_controller = obs_studio_controller
+
+    def get_play_video_action(
+        self,
+        track: Any,
+        current_offset: Callable[[], float],
+    ) -> Tuple[Action, str]:
+        path = os.path.join(self.root, track['obs_video'])
+        preload(path)
```

Comment:
I don't think we should be attempting to game the cache. Being a production software OBS should do sufficient caching on media sources so we should just let it handle it's own media. Also being a recording, the video doesn't actually have to sync with anything so the latency is not super critical.

Reply:
The video needs to sync with the overlay, at least to some extent.

With the current way that this works, OBS cannot do any preloading itself as it doesn't know what file to play until it's told to actually play it. I did previously explore various things to tell OBS about a video file before playing it, however there didn't seem to be a good mechanism for doing so -- everything I tried resulted in OBS playing the file immediately.